src/examples/fake_device_demo.rb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift(File.expand_path('../../lib', __FILE__))
+
+require "storage/fake_probing.rb"
+require "storage/fake_device_factory.rb"
+
+FILENAME = "fake-devicegraphs"
+
+fake_probing = Yast::Storage::FakeProbing.new
+devicegraph = fake_probing.devicegraph
+factory = Yast::Storage::FakeDeviceFactory.new(devicegraph)
+factory.load_yaml_file("#{FILENAME}.yml")
+
+fake_probing.to_probed
+puts("Disks:")
+probed = Yast::Storage::StorageManager.instance.probed
+fake_probing.dump_disks(probed)
```

Comment:
I dislike this approach for the same reasons I disliked the original code of the proposal with that `PROPOSAL_BASE` and `PROPOSAL` constants.

In my opinion, a FooFactory should generate by its own means objects of type Foo.

With this approach, we generate a Devicegraph object and we store it "somewhere" (in the drawer with the label FAKE) and the we pass that object to a ad-hoc "factory" that is used to tweak the object.

To be honest, I find the all `FakeProbing` class a little but cumbersome.

I would have expected the code on this file to look more functional. In Ruby, methods with input/output parameters or that simply leave the result elsewhere else instead of returning it feel wrong... and are usually hard to test. This looks way more ruby to me.
```ruby
#!/usr/bin/env ruby

$LOAD_PATH.unshift(File.expand_path('../../lib', __FILE__))

require "storage/fake_device_factory"
require "storage/debug_devicegraph"
using Yast::Storage::DebugDevicegraph

FILENAME = "fake-devicegraphs"
factory = Yast::Storage::FakeDeviceFactory.new
devicegraph = factory.load_yaml_file("#{FILENAME}.yml")

storage = StorageManager::create_fake_instance
storage.probed = devicegraph

storage.probed.dump_disks
storage.probed.write_graphviz("#{FILENAME}.gv")
system("dot -Tpng <#{FILENAME}.gv >#{FILENAME}.png")
system("display #{FILENAME}.png")

# Clean up
File.delete("#{FILENAME}.gv")
File.delete("#{FILENAME}.png")
```

Apart from being "more Ruby", I think it's more obvious.